specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,764 @@
+that will include any component adding to the AppNotification UI.
+
+**WinAppSDK 1.2 \<toast\> Schema**:
+
+```c#
+<toast
+    launch? = string
+    duration? = "long" | "short"
+    displayTimeStamp? = string
+    scenario? = "reminder" | "alarm" | "incomingCall" | "urgent"
+    useButtonStyle? = boolean>
+      <!-- Child elements -->
+    visual, audio?, actions?
+</toast>
+```
+
+These attributes are abstracted away through the AppNotificationContent component APIs:
+
+-   AddArgument(String key)
+-   AddArgument(String key, T value)
+    -   string
+    -   bool
+    -   integer
+    -   double
+    -   float
+-   SetDuration(enum Duration)
+    -   Duration::Short
+    -   Duration::Long
+-   SetUseButtonStyle(Boolean usesButtonStyle)
+-   SetTimeStamp(String timeStamp)
+    -   Must be in ISO 8601 format
+        -   Example: 2022-06-19T08:30:00Z
+-   SetScenario(ScenarioType scenarioType)
+    -   ScenarioType::Reminder
+    -   ScenarioType::Alarm
+    -   ScenarioType::IncomingCall
+    -   ScenarioType::Urgent
+
+All the UI components are able to be appended through the AppNotificationContent component APIs:
+
+-   AddText(Text text)
+-   AddButton(Button button)
+-   AddImage(Image image)
+-   AddAudio(Audio audio)
+-   AddTextBox(TextBox textBox)
+-   AddSelectionMenu(SelectionMenu selectionMenu)
+
+Below is an example usage:
+
+```c#
+AppNotificationContent()
+    .AddArgument(L"launch")
+    .SetDuration(Duration::Long)
+    .SetTimeStamp(L"2022-06-19T08:30:00Z")
+    .SetScenarioType(ScenarioType::IncomingCall)
+        .AddText(Text(L"Incoming Call")
+            .SetIsCallScenario(true))
+        .AddImage(Image(LR"(https://unsplash.it/100?image=883)")
+            .SetUsesCropCircle(true))
+        .AddButton(Button(L"Text Reply")
+            .SetIconUri(LR"(Assets/Icons/message.png)")
+            .AddArgument(L"action", L"textReply")
+            .AddArgument(L"callId", 938163))
+        .AddButton(Button(L"Reminder")
+            .SetIconUri(LR"(Assets/Icons/reminder.png)")
+            .AddArgument(L"action", L"reminder")
+            .AddArgument(L"callId", 938163))
+        .AddButton(Button(L"Ignore")
+            .SetIconUri(LR"(Assets/Icons/cancel.png)")
+            .AddArgument(L"action", L"ignore")
+            .AddArgument(L"callId", 938163))
+        .AddButton(Button(L"Answer")
+            .SetIconUri(LR"(Assets/Icons/telephone.png)")
+            .AddArgument(L"action", L"answer")
+            .AddArgument(L"callId", 938163))
+    .Show();
+```
+
+Example result:
+
+![AppNotification IncomingCall Example](AppNotificationIncomingCallExample.png)
+
+If the developer wants the XML payload to add more components, AppNotificationContent::GetXml will
+return the current version of their AppNotificationContent in a string format.
+
+```c#
+winrt::hstring xmlPayload{ AppNotificationContent()
+    .AddArgument(L"launch")
+        .AddText(Text(L"Content"))
+    .GetXml() };
+```
```

Comment:
Maybe it would be a good idea to add a code block for what GetXml() is returning in this case? So readers can also compare the XML to your example usage above.

Reply:
I agree. In fact, every example should have the resulting XML. I plan to add this the next iteration :)